### PR TITLE
server: Validate that only allowed parameters are used

### PR DESCRIPTION
**Pull request by supertassu**
Since we use Tekton provided pipelines[0], add a hardening step to
ensure that all pipeline parameters have been verified by us to be safe
to use.

[0]: https://github.com/toolforge/buildservice/blob/main/deploy/base/task-combined.yaml

### CODE REVIEW

**Comment by supertassu** [comment]
apparently I don't have enough access to this repo to use the request reviewers button, so: hi @david-caro @aborrero, could you review this PR?

File: pkg/server/validateadmission.go [hunk 1]
```diff
@@ -92,11 +92,30 @@ func (p *PipelineRunAdmission) HandleAdmission(review *admissionv1.AdmissionRevi
...
 				Result: &metav1.Status{
-					Message: fmt.Sprintf("Builder (gotten %s) does not match AllowedBuilders (expected %s).", param.Name, buildstr),
+					Message: fmt.Sprintf("Builder (gotten %s) does not match AllowedBuilders (expected %s).", param.Value.StringVal, buildstr),
 				},
 			}
 			return nil
+		case "SOURCE_URL":
+			// permitted parameter, but not validated
+		case "USER_ID":
+			// permitted parameter, but not validated (TODO: should it?)
```
Comment (david-caro) [inline pkg/server/validateadmission.go:102]: I was thinking the same, but in order to not allow passing it we have to use a mutator right?

File: pkg/server/server_test.go [hunk 2]
```diff
@@ -237,3 +242,29 @@ func TestHookAllowsGoodUserGoodDomainWithHTTPSProtocol(t *testing.T) {
...
+		},
+	})
+	requestString := string(encodeRequest(&badAdmissionReview))
+	myr := strings.NewReader(requestString)
+	r, _ := http.Post(server.URL, "application/json", myr)
+	review := decodeResponse(r.Body)
+	t.Log(review.Response)
+	if review.Response.Allowed {
+		t.Error("Allowed pipelinerun that should not have been allowed!")
+	}
```
Comment (david-caro) [inline pkg/server/server_test.go:269]: hmm... can we validate that it's complaining about the parameter not being valid and not something else? (currently that's what's expected and what happens, but if we add any other validation that might not be the case anymore and this test would need changing).

**Review by david-caro — COMMENTED** [review]
Got a comment and a question, but yes, it's a good idea (imo)

File: pkg/server/server_test.go (hunk 2 above)
Comment (supertassu) [reply pkg/server/server_test.go:269]: We can, done

File: pkg/server/validateadmission.go (hunk 1 above)
Comment (supertassu) [reply pkg/server/validateadmission.go:102]: Correct, I think that's out of scope for this patch

**Review by david-caro — APPROVED** [review]
Neat!

**Pull request merged**